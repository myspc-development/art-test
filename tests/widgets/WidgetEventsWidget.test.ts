import fs from 'fs';
import path from 'path';
import Ajv from 'ajv';
const ajv = new Ajv({ allErrors: true, strict: false });

const name = 'WidgetEventsWidget';
const rel = 'WidgetEventsWidget';
const schemaPath = path.join(process.cwd(), 'widgets', rel + '.schema.json');
const schema = JSON.parse(fs.readFileSync(schemaPath, 'utf8'));

test(`${name} schema validates minimal props`, () => {
  const validate = ajv.compile(schema);
  const ok = validate({});
  expect(ok).toBe(true);
});

const filePath = path.join(process.cwd(), 'widgets', rel + '.php');
test(`${name} PHP widget snapshot`, () => {
  const firstLine = fs.readFileSync(filePath, 'utf8').split('\n')[0];
  expect(firstLine).toMatchSnapshot();
});
