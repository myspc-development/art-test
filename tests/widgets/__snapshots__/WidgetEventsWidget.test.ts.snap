// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetEventsWidget PHP widget snapshot 1`] = `"<?php"`;
